.github/workflows/wheels.yml

```diff
@@ -1,47 +1,89 @@
-name: Python publication
-
 on:
   release:
     # run only on published, not for drafts
     types: [published]
+  pull_request:
+  workflow_dispatch:
+
+env:
+  flags: --release --out dist --manifest-path pineappl_py/Cargo.toml
 
 jobs:
-  deploy-linux:
+  linux:
     runs-on: ubuntu-latest
-    # a dedicated container is needed, so maturin-action is not an option, see pineappl_py/package/README.md for details
-    container:
-      image: docker://ghcr.io/n3pdf/maturin:0.2.1
+    strategy:
+      matrix:
+        target: [x86_64]
     steps:
-      - name: Compile and Publish wheels
-        working-directory: /home/pineappl
-        env:
-          HOME: /home/pineappl
-        run: |
-          ~/maturin publish --skip-existing --username __token__ --password ${{ secrets.PYPI_TOKEN }} --interpreter 3.7 3.8 3.9 3.10 pypy3.7
+      - uses: actions/checkout@v3
+      - name: Build wheels
+        uses: PyO3/maturin-action@v1
+        with:
+          target: ${{ matrix.target }}
+          args: ${{ env.flags }} --interpreter 3.7 3.8 3.9 3.10 3.11 pypy3.8
```

Comment:
The `interpreter` arguments are a few, can't we define a variable for them? Don't you believe they could be shared across different platforms? (at least the CPython versions, exclude `pypy`, for which one wheel is definitely enough, since we do not have any evidence for anyone using it)

Reply:
In particular, if you want to have a list, and not a single value, you can use `env` instead of `matrix`, as it was done in 03616cc8bfe8993b28b85a876f479fdfa38c007b

Reply:
Since by now you're not using `matrix` in any job, you won't need all the `fromJson` alchemy, but just define the string as you're using it (the exact same characters) and use the `env` context to expand:
```
${{ env.python_versions }}
```

Since contexts are expanded by the workflow runner, and not by the shell, there is no doubt it will be expanded before execution.

Reply:
I wanted to see whether passing multiple versions is possible at all (it is!).

Reply:
After our (private) conversation we decide to go back to `--find-interpreter` which selects all available Python versions.